### PR TITLE
Fix connections in ASMs3D

### PR DESCRIPTION
**Pull request by akva2**
This fixes oriented connections in ASMs3D when n != m.

Also adds test cases for all orientations, as well as a simple test where n != m.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Comment by akva2** [comment]
jenkins build this with downstreams please

File: src/ASM/Test/TestASMs3D.C [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+
+TEST_P(TestASMs3D, Connect)
+{
+  DummySIM<SIM3D> sim;
+  std::stringstream str;
+  str << "src/ASM/Test/refdata/DomainDecomposition_MPI_3D_4_orient";
+  str << GetParam() << ".xinp";
+  ASSERT_TRUE(sim.read(str.str().c_str()));
+  ASSERT_TRUE(sim.preprocess());
```
Comment (kmokstad) [inline src/ASM/Test/TestASMs3D.C]: You don't need to execute the preprocess() method to only test the connection stuff. It suffice to invoke createFEMmodel(). Then the DummySIM class is not needed either (can use SIM3D directly).

Comment (akva2) [reply src/ASM/Test/TestASMs3D.C]: indeed. i somehow was under the impression the method was protected.

**Comment by akva2** [comment]
jenkins build this with downstreams please

File: src/ASM/Test/TestASMs3D.C [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+//! \date Feb 14 2018
+//!
+//! \author Arne Morten Kvarving / SINTEF
+//!
+//! \brief Tests for structured 3D spline FE models.
+//!
+//==============================================================================
+
+#include "ASMs3D.h"
+#include "IntegrandBase.h"
```
Comment (kmokstad) [inline src/ASM/Test/TestASMs3D.C]: Remove IntegrandBase.h no longer used

Comment (akva2) [reply src/ASM/Test/TestASMs3D.C]: removed.

**Comment by VikingScientist** [comment]
More tests can be found at [vikingscientist / fix_s3d_connections_more_tests](https://github.com/VikingScientist/IFEM/tree/fix_s3d_connections_more_tests)

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Comment by akva2** [comment]
jenkins build this please

**Comment by VikingScientist** [comment]
Looks good :+1:

**Pull request merged**